app/components/application/AppName/AppName.tsx

```diff
@@ -39,6 +39,10 @@ export default function AppName({ id, name }: AppNameProps) {
     }
   }
 
+  useEffect(() => {
```

Comment:
This is used to sync props to the state. The state becomes stale if you switch between apps. The app name doesn't get updated because of the way the state is initialized.